e2e/app/component/select-menu.ts

```diff
@@ -98,17 +104,19 @@ export default class SelectMenu {
   }
 
   private dropdownXpath(): string {
-    if (this.label === undefined) {
-      return '//*[contains(concat(" ", normalize-space(@class), " "), " p-dropdown ")]';
+    if (this._label === undefined) {
```

Comment:
This looks really complex and prone to breakage by refactoring the dropdown component. I feel like there may be a way to simplify this by adding "handles" on the component (such as data tests IDs). Is that possible here?

The other problem is that the test class knows much more about the implementation than I'd be comfortable with if this is avoidable.

Reply:
Have you tried adding the `data-test-id`s yourself? I think that's a reasonable, low-risk thing you could do. Just add it where it's feasible and file tickets for those components that aren't test-friendly. Also a best practice doc on the test IDs and another one on xpath might be helpful.

The goal is for anyone looking at a failed test to be able to understand the code at a glance.